diff --git a/netbeans/bin/netbeans b/netbeans/bin/netbeans
index 42c9808..c72d1b7 100755
--- a/netbeans/bin/netbeans
+++ b/netbeans/bin/netbeans
@@ -46,8 +46,8 @@ case "`uname`" in
         ;;
     *) 
         # set default userdir and cachedir on unix systems
-        DEFAULT_USERDIR_ROOT=${HOME}/.netbeans
-        DEFAULT_CACHEDIR_ROOT=${HOME}/.cache/netbeans
+        DEFAULT_USERDIR_ROOT=${SNAP_USER_DATA}
+        DEFAULT_CACHEDIR_ROOT=${SNAP_USER_COMMON}
         ;;
 esac
 
